### PR TITLE
Add a flag that'll allow u-root to start systemd

### PR DESCRIPTION
**Pull request by BigValen**
SystemD can't start if it spots that cgroups /proc files have been created. It also requires that init really has PID1.

### CODE REVIEW

File: cmds/init/init.go [hunk 1]
```diff
@@ -170,20 +171,31 @@ func main() {
...
-			cmd.Env = envs
-			cmd.Stdin = os.Stdin
-			cmd.Stderr = os.Stderr
-			cmd.Stdout = os.Stdout
-			if *test {
-				cmd.SysProcAttr = &syscall.SysProcAttr{Cloneflags: cloneFlags}
+
+			initFlags := cmdline.GetInitFlagMap()
+			// systemd gets upset when it discovers it isn't really process 1
+			if systemd, present := initFlags["systemd"]; !present || systemd == "0" {
```
Comment (hugelgupf) [inline cmds/init/init.go]: use https://golang.org/pkg/strconv/#ParseBool for systemd?

**Comment by insomniacslk** [comment]
Seems like this test was broken in https://github.com/u-root/u-root/commit/574f37e690229904d17b33937418cc436bcdcf7d#diff-6be6ebcd84c2967933b4952e16db1668

**Comment by insomniacslk** [comment]
I am making a separate PR to fix that linter thing then @BigValen  you can rebase on top of it

**Comment by insomniacslk** [comment]
I will merge this as soon as the merge commits are removed

**Pull request merged**